### PR TITLE
New Rule (Link): Microsoft Dynamics 365 form phishing

### PR DESCRIPTION
**Pull request by aidenmitchell**

### CODE REVIEW

**Comment by aidenmitchell** [comment]
Unfortunately, the form loads too slowly for LA to get a complete look at the contents, would have used CredPhish otherwise. Regardless, this should hold up with the initial NLU detection. Running hunts now.

File: detection-rules/link_dynamics_form.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  - "https://cofense.com/blog/compromised-microsoft-dynamic-365-customer-voice-account-used-for-phishing-attack/"
+type: "rule"
+severity: "high"
+source: |
+  type.inbound
+  and any(ml.nlu_classifier(body.plain.raw).intents,
+      .name != "benign"
+  )
+  and any(body.links, 
+      beta.linkanalysis(.).effective_url.domain.domain == "customervoice.microsoft.com"
```
Comment (jkamdjou) [inline detection-rules/link_dynamics_form.yml:14]: i'd worry a little bit that we'd be invoking LA a bit too much w/o additional signals

also i think we can short-circuit some of this if we first look for the shortened version in the body (as an OR probably)

File: detection-rules/link_dynamics_form.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: "Link: Microsoft Dynamics 365 form phishing"
+description: |
+  Email body is suspicious, and links to a Microsoft Dynamics form. Known phishing tactic.
+references:
+  - "https://cofense.com/blog/compromised-microsoft-dynamic-365-customer-voice-account-used-for-phishing-attack/"
+type: "rule"
+severity: "high"
+source: |
+  type.inbound
+  and any(ml.nlu_classifier(body.plain.raw).intents,
```
Comment (jkamdjou) [inline detection-rules/link_dynamics_form.yml:10]: checkout usages of `coalesce` in other rules

**Comment by aidenmitchell** [comment]
@jkamdjou I went with `concat` instead of `coalesce`, NLU wasn't flagging our sample otherwise.

**Comment by aidenmitchell** [comment]
Hoping we can OCR the webpage screenshot and run it through NLU too, just to have a more confident detection.

File: detection-rules/link_dynamics_form.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          // confirm it is a form
+          and any(beta.linkanalysis(.).final_dom.links, .href_url.domain.domain == "cdn.forms.office.net")
+          and any(file.explode(beta.linkanalysis(.).screenshot),
+              any(ml.nlu_classifier(.scan.ocr.raw).intents,
+                  .name == "cred_theft" and
+                  .confidence == "high"
+              )
+          )
+      )
+  )
```
Comment (jkamdjou) [inline detection-rules/link_dynamics_form.yml:26]:
```suggestion
  )
  
  // unsolicited
  and (
          (
              sender.email.domain.root_domain in $free_email_providers
              and sender.email.email not in $recipient_emails
          )
          or (
              sender.email.domain.root_domain not in $free_email_providers
              and sender.email.domain.domain not in $recipient_domains
          )
  )
```

i think this should help reduce the number of messages we have to pass to NLU + LA, and hopefully not impact efficacy?

File: detection-rules/link_dynamics_form.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+tags: 
+  - "Suspicious links"
+  - "Suspicious body"
```
Comment (jkamdjou) [inline detection-rules/link_dynamics_form.yml:29]:
```suggestion
```

**Comment by aidenmitchell** [comment]
LGTM!

**Review by jkamdjou — APPROVED** [review]

**Pull request merged**